azext_edge/edge/providers/orchestration/deletion.py

```diff
@@ -0,0 +1,194 @@
+
+        self._render_progress = not no_progress
+        self._live = Live(None, transient=False, refresh_per_second=8, auto_refresh=self._render_progress)
+        self._progress_bar = Progress(
+            SpinnerColumn(),
+            *Progress.get_default_columns(),
+            "Elapsed:",
+            TimeElapsedColumn(),
+            transient=False,
+        )
+        self._progress_shown = False
+
+    def do_work(self, confirm_yes: Optional[bool] = None, force: Optional[bool] = None):
+        self._display_resource_tree()
+
+        should_delete = True
+        if not confirm_yes:
+            should_delete = Confirm.ask("Continue?")
+
+        if not should_delete:
+            logger.warning("Deletion cancelled.")
+            return
+
+        self._process(force=force)
+
+    def _display_resource_tree(self):
+        if self._render_progress:
+            print(self.resource_map.build_tree())
+
+    def _render_display(self, description: str):
+        if self._render_progress:
+            grid = Table.grid(expand=False)
+            grid.add_column()
+            grid.add_row(NewLine(1))
+            grid.add_row(description)
+            grid.add_row(NewLine(1))
+            grid.add_row(self._progress_bar)
+
+            if not self._progress_shown:
+                self._task_id = self._progress_bar.add_task(description="Work.", total=None)
+                self._progress_shown = True
+            self._live.update(grid, refresh=True)
+
+            if not self._live.is_started:
+                self._live.start(True)
+
+    def _stop_display(self):
+        if self._render_progress and self._live.is_started:
+            if self._progress_shown:
+                self._progress_bar.update(self._task_id, description="Done.")
+                sleep(0.5)
+            self._live.stop()
+
+    def _process(self, force: bool = False):
+        todo_extensions = self.resource_map.extensions
+        todo_custom_locations = self.resource_map.custom_locations
+        todo_resource_sync_rules = []
+        todo_resources = []
+
+        for cl in todo_custom_locations:
+            todo_resource_sync_rules.extend(self.resource_map.get_resource_sync_rules(cl.resource_id))
+            todo_resources.extend(self.resource_map.get_resources(cl.resource_id))
+
+        batched_work = self._batch_resources(
+            resources=todo_resources,
+            resource_sync_rules=todo_resource_sync_rules,
+            custom_locations=todo_custom_locations,
+            extensions=todo_extensions,
+        )
+        if not batched_work:
+            logger.warning("Nothing to delete :)")
+            return
+
+        if not force:
+            if not self.resource_map.connected_cluster.connected:
+                logger.warning(
+                    "Deletion cancelled. The cluster is not connected to Azure. "
+                    "Use --force to continue anyway. Not recommended."
```

Comment:
do you know what happens if a cluster is not deleted correctly? (aka will people get mad at us)

Reply:
You can keep running the delete operation. However if a cluster is not online it will most likely fail.